### PR TITLE
Increase detail of information provided to statusCallback

### PR DESCRIPTION
**Pull request by joshjung**
This includes several additions which we needed for more detailed statistical analysis of load testing. We use the updated code to allow us to export a CSV spreadsheet of every call and its elapsed time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -617,9 +617,19 @@ Example:
...
+Execution this function after every request operation completes. Provides immediate access to test results while the
+test batch is still running. This can be used for more detailed custom logging or developing your own spreadsheet or
+statistical analysis of results.
+
+The results and error passed to the callback are in the same format as the results passed to the final callback.
+ 
+In addition, the following three properties are added to the `result` OR `error` object:
+
+- `requestElapsed`: time in milliseconds it took to complete this individual request.
+- `requestIx`: 0-based index of this particular request in the sequence of all requests to be made.
```
Comment (alexfernandez) [inline README.md]: Please don't use `Ix` but the full word `Index`.

File: lib/loadtest.js [hunk 2]
```diff
@@ -248,7 +248,20 @@ function HttpClient(operation, params)
 			{
 				log.debug('Connection %s ended', id);
 			}
-			operation.latency.end(id, errorCode);
+
+			var elapsed = operation.latency.end(id, errorCode);
+			var ix = operation.latency.getRequestIndex(id);
```
Comment (alexfernandez) [inline lib/loadtest.js]: Please don't use `ix` but the full word `index`.

File: lib/loadtest.js [hunk 3]
```diff
@@ -248,7 +248,20 @@ function HttpClient(operation, params)
...
+
+			if (error && typeof error === 'object') {
+				error.requestElapsed = elapsed;
+				error.requestIx = ix;
```
Comment (alexfernandez) [inline lib/loadtest.js]: Please don't use `Ix` but the full word `Index`.

File: lib/timing.js [hunk 4]
```diff
@@ -41,11 +41,15 @@ exports.Latency = function(options, callback)
 	var totalTime = 0;
 	var totalErrors = 0;
 	var maxLatencyMs = 0;
+	var minLatencyMs = 999999;
 	var histogramMs = {};
 	var errorCodes = {};
 	var running = true;
 	var totalsShown = false;
 
+	var requestIx = 0;
```
Comment (alexfernandez) [inline lib/timing.js]: Please don't use `Ix` but the full word `Index`.

**Comment by joshjung** [comment]
Hey there, I fixed the Ix references and then did a commit amend. Let me know what you think!

**Comment by alexfernandez** [comment]
@joshjung I have changed the order of parameters in 2.0.0 to conform to Node.js standards, I think it is much better now.

**Pull request merged**